runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SplittableParDoProcessKeyedElementsOp.java

```diff
@@ -137,6 +142,14 @@ public void open(
             isBounded,
             pipelineOptions);
 
+    if (this.ses.get() == null) {
+      this.ses.compareAndSet(
+          null,
+          Executors.newScheduledThreadPool(
+              Runtime.getRuntime().availableProcessors(),
```

Comment:
ditto since this one is just used by a single processfn

Reply:
changed back to newSingleThreadScheduledExecutor and removed atomicreference